lib/index.js

```diff
@@ -49,10 +49,28 @@ class SpikeCollections {
         cwd: compiler.options.context,
         nodir: true,
         realpath: true,
-        ignore: compiler.options.spike.ignore
+        ignore: this.util.getSpikeOptions().ignore
       })
     })
 
+    // all files as a straight array, no categories
+    this.allFiles = reduce(this.files, (m, v) => m.push(...v) && m, [])
+
+    // add loader alias so that frontmatter loader can be resolved
+    const resolveLoader = compiler.options.resolveLoader
+    if (!resolveLoader.alias) resolveLoader.alias = {}
+    resolveLoader.alias.frontmatter = path.join(__dirname, 'frontmatter_loader.js')
+
+    // add frontmatter loader with pattern converted to regex
+    compiler.options.module.rules.push({
+      test: this.util.pathsToRegex(this.allFiles),
+      use: [{
+        loader: 'frontmatter',
+        // NOTE: this *could* be configurable
+        options: { _spikeExtension: 'html' }
+      }]
+    })
```

Comment:
To add to this, here's the spot in webpack's source where the rules are "sealed":

https://github.com/webpack/webpack/blob/760c2471682e3438d5323d042c30392f5f625a70/lib/NormalModuleFactory.js#L46

The `RuleSet` class basically takes the rules options and validates and normalizes them, returning a clean and valid copy. Adding additional rules or modifying rules after this happens will result in no effect, since the normal module factory already is holding on to the validated and sanitized copy.

This means the you must make any rule modifications *before* [the `normal-module-factory` hook](https://webpack.js.org/api/plugins/#the-normalmodulefactory) fires. The `make` hook goes off well after this, which is why this code needed to be changed.